redis/src/lib.rs

```diff
@@ -75,3 +75,42 @@ impl bb8::ManageConnection for RedisConnectionManager {
         false
     }
 }
+
+/// A `bb8::ManageConnection` for `redis::Client::get_async_connection`.
+#[derive(Clone, Debug)]
+pub struct RedisPubSubConnectionManager {
+    client: Client,
+}
+
+impl RedisPubSubConnectionManager {
+    /// Create a new `RedisPubSubConnectionManager`.
+    /// See `redis::Client::open` for a description of the parameter types.
+    pub fn new<T: IntoConnectionInfo>(info: T) -> Result<RedisPubSubConnectionManager, RedisError> {
+        Ok(RedisPubSubConnectionManager {
+            client: Client::open(info.into_connection_info()?)?,
+        })
+    }
+}
+
+#[async_trait]
+impl bb8::ManageConnection for RedisPubSubConnectionManager {
+    type Connection = PubSub;
+    type Error = RedisError;
+
+    async fn connect(&self) -> Result<Self::Connection, Self::Error> {
+        self.client
+            .get_async_connection()
+            .await
+            .map(|conn| conn.into_pubsub())
+    }
+
+    async fn is_valid(&self, conn: &mut Self::Connection) -> Result<(), Self::Error> {
+        let test_sub_name = "__test__-sub-from-bb8";
+        conn.subscribe(test_sub_name).await?;
+        conn.unsubscribe(test_sub_name).await
```

Comment:
this feels hackish, but there is not much that can be done on a pubsub connection, basic CMD aren't available for example